R/Model-class.R

```diff
@@ -2653,6 +2653,92 @@ OneParExpNormalPrior <- function(skel_probs,
   )
 }
 
+# OneParExpExpPrior ----
+
+## class ----
+
+#' `OneParExpExpPrior`
+#'
+#' @description `r lifecycle::badge("experimental")`
+#'
+#' [`OneParExpExpPrior`] is the class for a standard CRM with an exponential prior on
+#' the log power parameter for the skeleton prior probabilities. It is an
+#' implementation of a version of the one-parameter CRM (O’Quigley et al. 1990).
+#'
+#' @slot skel_fun (`function`)\cr function to calculate the prior DLT probabilities.
+#' @slot skel_fun_inv (`function`)\cr inverse function of `skel_fun`.
+#' @slot skel_probs (`numeric`)\cr skeleton prior probabilities. This is a vector
+#'   of unique and sorted probability values between 0 and 1.
+#' @slot lambda (`number`)\cr rate parameter of prior exponential distribution for theta.
+#'
+#' @aliases OneParExpExpPrior
+#' @export
+#'
+.OneParExpExpPrior <- setClass(
+  Class = "OneParExpExpPrior",
+  slots = c(
+    skel_fun = "function",
+    skel_fun_inv = "function",
+    skel_probs = "numeric",
+    lambda = "numeric"
+  ),
+  contains = "GeneralModel",
+  validity = v_model_one_par_exp_exp_prior
+)
+
+## constructor ----
+
+#' @rdname OneParExpExpPrior-class
+#'
+#' @param skel_probs (`numeric`)\cr skeleton prior probabilities. This is a vector
+#'   of unique and sorted probability values between 0 and 1.
+#' @param dose_grid (`numeric`)\cr dose grid. It must be must be a sorted vector
+#'   of the same length as `skel_probs`.
+#' @param lambda (`number`)\cr rate parameter of prior exponential distribution for theta.
+#'
+#' @export
+#' @example examples/Model-class-OneParExpExpPrior.R
+#'
+OneParExpExpPrior <- function(skel_probs,
+                              dose_grid,
+                              lambda) {
+  assert_probabilities(skel_probs, unique = TRUE, sorted = TRUE) # So that skel_fun_inv exists.
+  assert_numeric(dose_grid, len = length(skel_probs), any.missing = FALSE, unique = TRUE, sorted = TRUE)
+  assert_number(lambda, lower = .Machine$double.xmin, finite = TRUE)
+
```

Comment:
Assertion for `lambda` is actually not needed here (we have it in class validation function). Same for `OneParExpNormalPrior` class and `sigma2` param.

Reply:
will remove both